### PR TITLE
feat(pf4): add duallist component

### LINKED ISSUE #358: dual-list-select is not available in PF4
**Scope**: PF4 dual-list-select

**Description**
Attempted to use `dual-list-select` in a PF4 app and hit the following error:
dual-list-select

**Schema**

Use the example from here:
https://data-driven-forms.org/others/miq-components#duallistselect

### PR DESCRIPTION
**Pull request by rvsia**
closes #358 

Add Dual List 'placeholder' component

image

duallistselect

@chambridge  @Hyperkid123 

What do you think about the design?
Will you also need async filtering?

todo:
- [x] docs
- [x] tests
- [x] ~~async filtering for options~~ (in the future)

### CODE REVIEW

**Comment by chambridge** [comment]
@rvsia Looks good to me.

Can you clarify async filtering? are you meaning in order to make http requests on the dataset? That is a nice to have, but I think we can make it work if we don't have that.

I think what you show above would meet our needs.

**Comment by rvsia** [comment]
@chambridge Yeah, I am talking about filtering using API's calls. Our API has a maximum limit of 100 returned items, so I can imagine that if users has more than 100 OpenShift labels, it would be problem to show them all without asynchronous filtering.

**Comment by chambridge** [comment]
That would certainly be nice, but since the API is building the data driven forms it in theory it could collect them (there is just a potential performance impact) to collect all the labels when settings is called.

**Comment by Hyperkid123** [comment]
@rvsia we can add async filtering later. We might find more requirements after this component is published and used.

**Comment by rvsia** [comment]
@chambridge Do you want to test this PR too?

File: packages/pf4-component-mapper/src/form-fields/dual-list-select.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.ddorg__pf4-component-mapper__dual-list-select-toolbar {
+    height: auto !important;
+    padding: 0px;
+}
+
+.ddorg__pf4-component-mapper__dual-list-select-toolbar-content {
+    padding: 0px 0px 16px 0px !important;
```
Comment (Hyperkid123) [inline packages/pf4-component-mapper/src/form-fields/dual-list-select.scss]: you can us pf modifier classes like `pf-p-m-0` `pf-u-pr-md` etc. So we can remove the !important nonsense :smile:

Comment (rvsia) [reply packages/pf4-component-mapper/src/form-fields/dual-list-select.scss]: let's them make the nonsense :see_no_evil:

**Comment by rvsia** [comment]
> @rvsia can you add the addons (just the classes we used) to the docs as well?

They are already there

**Pull request merged**